lib/supavisor/db_handler.ex

```diff
@@ -135,166 +139,64 @@ defmodule Supavisor.DbHandler do
-      {_, :authentication_sasl, nonce} ->
+      {:authentication_sasl, nonce} ->
         {:keep_state, %{data | nonce: nonce}}
 
-      {_, :authentication_server_first_message, server_proof} ->
+      {:authentication_server_first_message, server_proof} ->
         {:keep_state, %{data | server_proof: server_proof}}
 
-      {_, :authentication_md5} ->
+      %{authentication_server_final_message: _server_final} ->
+        :keep_state_and_data
+
+      %{authentication_ok: true} ->
+        :keep_state_and_data
+
+      :authentication ->
+        :keep_state_and_data
+
+      :authentication_md5 ->
         {:keep_state, data}
 
       {:error_response, ["SFATAL", "VFATAL", "C28P01", reason, _, _, _]} ->
-        tenant = Supavisor.tenant(data.id)
-
-        for node <- [node() | Node.list()] do
-          :erpc.cast(node, fn ->
-            Cachex.del(Supavisor.Cache, {:secrets, tenant, data.user})
-            Cachex.del(Supavisor.Cache, {:secrets_check, tenant, data.user})
-
-            Registry.dispatch(Supavisor.Registry.TenantClients, data.id, fn entries ->
-              for {client_handler, _meta} <- entries,
-                  do: send(client_handler, {:disconnect, reason})
-            end)
-          end)
-        end
-
-        Supavisor.stop(data.id)
-        Logger.error("DbHandler: Auth error #{inspect(reason)}")
+        Logger.error("ProxyDb: Auth error #{inspect(reason)}")
         {:stop, :invalid_password, data}
 
       {:error_response, error} ->
-        Logger.error("DbHandler: Error auth response #{inspect(error)}")
+        Logger.error("ProxyDb: Error auth response #{inspect(error)}")
         {:keep_state, data}
 
-      {:ready_for_query, ps, db_state} ->
+      {:ready_for_query, acc} ->
+        ps = acc.ps
+
         Logger.debug(
-          "DbHandler: DB ready_for_query: #{inspect(db_state)} #{inspect(ps, pretty: true)}"
+          "ProxyDb: DB ready_for_query: #{inspect(acc.db_state)} #{inspect(ps, pretty: true)}"
         )
 
-        Supavisor.set_parameter_status(data.id, ps)
+        if data.proxy do
+          bin_ps = Server.encode_parameter_status(ps)
+          send(data.caller, {:parameter_status, bin_ps})
+        else
+          Supavisor.set_parameter_status(data.id, ps)
+        end
 
         {:next_state, :idle, %{data | parameter_status: ps},
          {:next_event, :internal, :check_buffer}}
 
       other ->
-        Logger.error("DbHandler: Undefined auth response #{inspect(other)}")
+        Logger.error("ProxyDb: Undefined auth response #{inspect(other)}")
         {:stop, :auth_error, data}
     end
   end
 
-  def handle_event(:internal, :check_buffer, :idle, %{reply: {from, pid}} = data) do
-    :ok = Helpers.controlling_process(data.sock, pid)
-    reply = {:reply, from, {:ok, data.sock}}
-    {:next_state, :busy, %{data | reply: nil}, reply}
+  def handle_event(:internal, :check_buffer, :idle, %{reply: from} = data) when from != nil do
+    Logger.debug("DbHandler: Check buffer")
```

Comment:
It should be `ProxyDb` or `DbHandler`? Because it is mixed in the different log entries.

Reply:
DbHandler, ProxyDb is a legacy from the another handler